rootfs/registry/dockerclient.py

```diff
@@ -55,8 +54,8 @@ def publish_release(self, source, config, target, deis_registry):
     def build(self, source, config, repo, tag):
         """Add a "last-mile" layer of environment config to a Docker image for deis-registry."""
         check_blacklist(repo)
-        env = ' '.join("{}='{}'".format(
-            k, v.encode('unicode-escape').replace("'", "\\'")) for k, v in config.viewitems())
+        env = ' '.join("{}='{}'".format(k, v) for k, v in config.items())
```

Comment:
The unicode-escape encoder handles quotes and slashes and such perfectly for this case, except for single-quote. There's a lot of lore about this on StackOverflow and elsewhere. The `ENV` line in a Dockerfile needs to be escaped. Without explicit encoding we'll have problems using env vars that contain quotes and other such characters, but honestly I don't know how Python 3.5 treats this.